[BUG] Installation Script failure - ERROR: Package 'foo' requires a different Python: 3.12.3 not in '>=3.14'

## 🐛 Bug Description
A clear and concise description of what the bug is.

## 📋 Steps to Reproduce
Create a new directory and run `uv init` in that directory. Then run the vocalinux installer.
```bash
mkdir ~/foo; cd foo; uv init
$ curl -fsSL raw.githubusercontent.com/jatinkrmalik/vocalinux/main/install.sh -o /tmp/vl.sh && bash /tmp/vl.sh
```

<details><summary> Output from vocalinux installer: </summary>

```
$ curl -fsSL raw.githubusercontent.com/jatinkrmalik/vocalinux/main/install.sh -o /tmp/vl.sh && bash /tmp/vl.sh

  ▗▖  ▗▖ ▗▄▖  ▗▄▄▖ ▗▄▖ ▗▖   ▗▄▄▄▖▗▖  ▗▖▗▖ ▗▖▗▖  ▗▖
  ▐▌  ▐▌▐▌ ▐▌▐▌   ▐▌ ▐▌▐▌     █  ▐▛▚▖▐▌▐▌ ▐▌ ▝▚▞▘
  ▐▌  ▐▌▐▌ ▐▌▐▌   ▐▛▀▜▌▐▌     █  ▐▌ ▝▜▌▐▌ ▐▌  ▐▌
   ▝▚▞▘ ▝▚▄▞▘▝▚▄▄▖▐▌ ▐▌▐▙▄▄▖▗▄█▄▖▐▌  ▐▌▝▚▄▞▘▗▞▘▝▚▖

                    Voice Dictation for Linux

[INFO] Vocalinux Installer
[INFO] ==============================

[INFO] Running from local repository: /home/lastmile/foo
[INFO] Using virtual environment: /home/lastmile/foo/venv

[INFO] Detected: Ubuntu 24.04 (ubuntu family)

                 Interactive Installation Guide
                 ===============================

Welcome! This guided installation will help you set up Vocalinux
with the best options for your system.

All speech engines are 100% offline, local, and private.
Your voice data never leaves your computer.


============================================================
  Step 1: Your System
============================================================
Detected: Ubuntu 24.04
Hardware: Vulkan-compatible GPU detected - Great performance with whisper.cpp Vulkan


============================================================
  Step 2: Choose Speech Recognition Engine
============================================================

  ┌─────────────────────────────────────────────────────────────┐
  │  1. WHISPER.CPP  ★ RECOMMENDED                              │
  │     • Fastest, most accurate, works with any GPU            │
  │     • Supports NVIDIA (CUDA), AMD, Intel (Vulkan)           │
  │     • CPU-only mode available for older systems             │
  │     • Models: tiny (39MB) to large (1.5GB)                  │
  │     • 99+ languages with auto-detection                     │
  └─────────────────────────────────────────────────────────────┘

  ┌─────────────────────────────────────────────────────────────┐
  │  2. WHISPER (OpenAI)                                        │
  │     • PyTorch-based, high accuracy                          │
  │     • Only supports NVIDIA GPUs (CUDA)                      │
  │     • Larger download (~2GB with CUDA)                      │
  │     • Good for development/research                         │
  └─────────────────────────────────────────────────────────────┘

  ┌─────────────────────────────────────────────────────────────┐
  │  3. VOSK                                                    │
  │     • Lightweight and fast                                  │
  │     • Works on older/low-RAM systems                        │
  │     • ~40MB download                                        │
  │     • Good for basic dictation needs                        │
  └─────────────────────────────────────────────────────────────┘

  → Recommendation: whisper.cpp (best performance for your hardware)

Choose engine [1-3] (default: 1):

============================================================
  Step 3: Choose Whisper.cpp Backend
============================================================

Whisper.cpp can use different backends for speech recognition:

  ┌─────────────────────────────────────────────────────────────┐
  │  1. GPU (Vulkan/CUDA)  ★ RECOMMENDED                        │
  │     • Fastest performance with GPU acceleration             │
  │     • Possible Vulkan GPU (will verify during build)                                   │
  │     • Requires building from source (takes ~2-5 min)        │
  └─────────────────────────────────────────────────────────────┘

  ┌─────────────────────────────────────────────────────────────┐
  │  2. CPU (Pre-built)                                         │
  │     • Works on all systems                                  │
  │     • Faster installation (no compilation)                  │
  │     • Good performance on modern CPUs                       │
  └─────────────────────────────────────────────────────────────┘

  → Recommendation: GPU backend for best performance
Choose backend [1-2] (default: 1):


============================================================
  Step 4: Model Download
============================================================

Speech recognition models can be downloaded now or later.

  1. Download now (recommended)
     • Faster first run - ready to use immediately
     • Offline capable right after install

  2. Download later
     • Smaller initial install
     • Models download automatically on first use

Download models now? [1-2] (default: 1):

============================================================
  Installation Summary
============================================================

  Speech Engine: Whisper.cpp (Recommended)
  Backend: GPU (Vulkan/CUDA)
  Note: GPU build will compile from source (2-5 minutes)
  Models: Download now (recommended)
  Install Location: /home/lastmile/foo

Press Enter to continue with installation, or Ctrl+C to cancel...

[INFO] Installing system dependencies...
[INFO] All required packages are already installed.
[INFO] Detected session type: x11
[INFO] Installing X11 text input tools...
[INFO] xdotool is already installed.
[INFO] Creating application directories...
[INFO] Detected Python version: 3.12
[INFO] Setting up Python virtual environment in /home/lastmile/foo/venv...
[INFO] Updating pip, setuptools, and wheel...
Requirement already satisfied: pip in ./venv/lib/python3.12/site-packages (24.0)
Collecting pip
  Using cached pip-26.0.1-py3-none-any.whl.metadata (4.7 kB)
Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (68.1.2)
Collecting setuptools
  Using cached setuptools-82.0.1-py3-none-any.whl.metadata (6.5 kB)
Requirement already satisfied: wheel in /usr/lib/python3/dist-packages (0.42.0)
Collecting wheel
  Using cached wheel-0.46.3-py3-none-any.whl.metadata (2.4 kB)
Requirement already satisfied: packaging>=24.0 in /usr/lib/python3/dist-packages (from wheel) (24.0)
Using cached pip-26.0.1-py3-none-any.whl (1.8 MB)
Using cached setuptools-82.0.1-py3-none-any.whl (1.0 MB)
Using cached wheel-0.46.3-py3-none-any.whl (30 kB)
Installing collected packages: wheel, setuptools, pip
  Attempting uninstall: wheel
    Found existing installation: wheel 0.42.0
    Not uninstalling wheel at /usr/lib/python3/dist-packages, outside environment /home/lastmile/foo/venv
    Can't uninstall 'wheel'. No files were found to uninstall.
  Attempting uninstall: setuptools
    Found existing installation: setuptools 68.1.2
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /home/lastmile/foo/venv
    Can't uninstall 'setuptools'. No files were found to uninstall.
  Attempting uninstall: pip
    Found existing installation: pip 24.0
    Uninstalling pip-24.0:
      Successfully uninstalled pip-24.0
Successfully installed pip-26.0.1 setuptools-82.0.1 wheel-0.46.3
[INFO] Virtual environment activated successfully.
[INFO] Created activation script: ./activate-vocalinux.sh
[INFO] Detected GI_TYPELIB_PATH: /usr/lib/x86_64-linux-gnu/girepository-1.0
[INFO] Installing Vocalinux...
Processing ./.
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
INFO: pip is looking at multiple versions of foo to determine which version is compatible with other requirements. This could take a while.
ERROR: Package 'foo' requires a different Python: 3.12.3 not in '>=3.14'
[ERROR] Failed to install Vocalinux.
[ERROR] Check the pip log for details: /tmp/tmp.WPnUTlV4gt/pip_log.txt
[ERROR] Failed to install Vocalinux package. Installation cannot continue.
```

</details>


## ✅ Expected Behavior
- Vocalinux installer should do the same thing regardless of which directory it's run in.
- Vocalinux installer should ask the user before modifying state in local directory, or (better yet) ask the user where they want to install vocalinux.

## ❌ Actual Behavior
- local directory state prevented vocalinux installation from succeeding


## 💻 Environment
- **OS**:
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.3 LTS
Release:        24.04
Codename:       noble
```

--- a/install.sh
+++ b/install.sh
@@ -364,8 +364,17 @@ ensure_git_installed() {
     print_success "git installed successfully!"
 }
 
-if [ -f "setup.py" ] || [ -f "pyproject.toml" ]; then
-    # Running from within the repo
+# Check if running from within the vocalinux repository
+# Validate that pyproject.toml/setup.py actually belongs to vocalinux before entering local repo mode
+IS_VOCALINUX_LOCAL=false
+if [ -f "pyproject.toml" ] && grep -q 'name = "vocalinux"' "pyproject.toml" 2>/dev/null; then
+    IS_VOCALINUX_LOCAL=true
+elif [ -f "setup.py" ] && grep -q "vocalinux" "setup.py" 2>/dev/null; then
+    IS_VOCALINUX_LOCAL=true
+fi
+
+if [ "$IS_VOCALINUX_LOCAL" = true ]; then
+    # Running from within the vocalinux repo
     INSTALL_DIR="$(pwd)"
     print_info "Running from local repository: $INSTALL_DIR"
     # Convert VENV_DIR to absolute path for wrapper scripts

